src/PowerShellEditorServices/Symbols/PesterDocumentSymbolProvider.cs

```diff
@@ -27,63 +27,104 @@ public class PesterDocumentSymbolProvider : FeatureProviderBase, IDocumentSymbol
+        /// </summary>
+        /// <param name="commandAst">the CommandAst to test</param>
+        /// <returns>true if the CommandAst represents a Pester command, false otherwise</returns>
+        private static bool IsPesterCommand(CommandAst commandAst)
+        {
+            if (commandAst == null)
+            {
+                return false;
+            }
 
-                    // The -Name parameter
-                    for (int i = 1; i < testAst.CommandElements.Count; i++)
-                    {
-                        CommandParameterAst paramAst = testAst.CommandElements[i] as CommandParameterAst;
-                        if (paramAst != null &&
-                            paramAst.ParameterName.Equals("Name", StringComparison.OrdinalIgnoreCase))
-                        {
-                            testNameParamIndex = i + 1;
-                            break;
-                        }
-                    }
+            // Ensure the first word is a Pester keyword
+            if (!(commandAst.CommandElements[0] is StringConstantExpressionAst pesterKeywordAst &&
+                PesterSymbolReference.PesterKeywords.ContainsKey(pesterKeywordAst.Value)))
+            {
+                return false;
+            }
 
-                    if (testNameParamIndex > testAst.CommandElements.Count - 1)
-                    {
-                        return null;
-                    }
+            // Ensure that the last argument of the command is a scriptblock
+            if (!(commandAst.CommandElements[commandAst.CommandElements.Count-1] is ScriptBlockExpressionAst))
+            {
+                return false;
+            }
 
-                    StringConstantExpressionAst stringAst =
-                        testAst.CommandElements[testNameParamIndex] as StringConstantExpressionAst;
+            return true;
+        }
+
+        /// <summary>
+        /// Convert a CommandAst known to represent a Pester command and a reference to the scriptfile
+        /// it is in into symbol representing a Pester call for code lens
+        /// </summary>
+        /// <param name="scriptFile">the scriptfile the Pester call occurs in</param>
+        /// <param name="pesterCommandAst">the CommandAst representing the Pester call</param>
+        /// <returns>a symbol representing the Pester call containing metadata for CodeLens to use</returns>
+        private static PesterSymbolReference ConvertPesterAstToSymbolReference(ScriptFile scriptFile, CommandAst pesterCommandAst)
+        {
+            string testLine = scriptFile.GetLine(pesterCommandAst.Extent.StartLineNumber);
+            string commandName = (pesterCommandAst.CommandElements[0] as StringConstantExpressionAst)?.Value;
 
-                    if (stringAst == null)
+            // Search for a name for the test
+            string testName = null;
+            for (int i = 1; i < pesterCommandAst.CommandElements.Count; i++)
```

Comment:
`StaticParameterBinder.BindCommand()` is excellent for this type of analysis.  Might not be available in PowerShell 3 though.

Reply:
Is there an easy way to find out? I don't really know where I could get a PowerShell v3 instance to run

Reply:
> Is there an easy way to find out?

It'll fail to compile when it tries to build using the v3 reference libs.  It won't show any syntax markers because of the compiler directives though, and that's a huge pain.

> I don't really know where I could get a PowerShell v3 instance to run

I just have a Windows 7 VM set up in my home lab that I manually upgraded from 2 to 3.  It's annoying, but works.

Reply:
@rjmholt I've got a Windows Server VM for each major version of PowerShell on my Azure subscription. They're all turned off and deallocated (saving that $$$) until needed.

Reply:
@tylerl0706 Excellent!

Reply:
If you need me to test something, I can 😄 just give me the script to run. Otherwise, tomorrow I'll add you to those VMs so we can just share them.